src/v/rpc/transport.cc

```diff
@@ -363,14 +363,23 @@ ss::future<> transport::do_dispatch_send() {
 }
 
 void transport::dispatch_send() {
-    ssx::spawn_with_gate(_dispatch_gate, [this]() mutable {
-        return ssx::ignore_shutdown_exceptions(do_dispatch_send())
-          .handle_exception([this](std::exception_ptr e) {
-              vlog(rpclog.info, "Error dispatching socket write:{}", e);
-              _probe->request_error();
-              fail_outstanding_futures();
-          });
-    });
+    if (!_dispatch_gate.try_enter()) {
+        return;
+    }
+    ssx::background = ssx::ignore_shutdown_exceptions(do_dispatch_send())
+                        .then_wrapped([this](ss::future<> fut) {
+                            if (fut.failed()) {
+                                vlog(
+                                  rpclog.info,
+                                  "Error dispatching socket write:{}",
+                                  fut.get_exception());
+                                _probe->request_error();
+                                fail_outstanding_futures();
+                            }
+                            // Now that we've handled errors it's safe to
+                            // release the gate.
+                            _dispatch_gate.leave();
+                        });
```

Comment:
there is a slight preference to use raii gate holder rather than manual enter/leave. you can move that into the `then_wrapped continuation as as capture`, or dangle it off the end in a finally block capture.

Reply:
Good suggestion. `dispatch_send` shouldn't throw, which is why I manually used try_enter and leave, but it seems we still can check if the gate is closed before holding the gate open.